peps/pep-0763.rst

```diff
@@ -0,0 +1,275 @@
+   distributions or wheels).
+
+   Example: ``requests-2.32.3-py3-none-any.whl``.
+
+2. **Versions**, which contain one or more files that share the same version.
+
+   Example: `requests v2.32.3 <https://pypi.org/project/requests/2.32.3/>`_.
+
+3. **Projects**, which contain one or more versions.
+
+   Example: `requests <https://pypi.org/project/requests>`_.
+
+This PEP proposes the following *deletion eligibility rules*:
+
+* A **file** is considered deletable if and only if it was uploaded to
+  PyPI less than 72 hours from the current time, **or** if it
+  has a :ref:`pre-release specifier <packaging:version-specifiers>`.
+* A **release** is considered deletable if and only if all of its
+  constituent files are deletable.
+* A **project** is considered deletable if and only if all of its
+  constituent releases are deletable.
+
+These rules are intentionally "telescoping": they allow new projects to be
+deleted entirely, and allow old projects to delete new files or releases,
+but do not allow old projects to delete old files or releases.
+
+This is intended to strike a balance between competing interests: brand new
+projects are unlikely to have significant community uptake and thus pose a
+minimal disruptive risk, while established projects (of any size)
+are more likely to have a "tail" of adopted versions. Their downstream users
+are not necessarily equipped to address the sudden deletion
+of a version, file, or the whole project.
+
+Implementation
+==============
+
+This PEP's implementation primarily concerns aspects of PyPI that are not
+standardized or subject to standardization, such as the web interface and
+signed-in user operations. As a result, this section describes its
+implementation in behavioral terms.
+
+Changes
+-------
+
+* Per the eligibility rules above, PyPI will reject web interface requests
+  (using an appropriate HTTP response code of its choosing) for
+  file, release, or project deletion if the respective object is not
+  eligible for deletion.
+* PyPI will amend its web interface to indicate a file/release/project's
+  deletion ineligibility, e.g. by styling the relevant UI elements as "inactive"
+  and making relevant bottoms/forms unclickable.
+
+Security Implications
+=====================
+
+This PEP does not identify any positive or negative security implications
+associated with proposed approach.
+
+How To Teach This
+=================
+
+This PEP suggests at least two pieces of public-facing material to help
+the larger Python packaging community (and its downstream consumers)
+understand its changes:
+
+* An announcement post on the `PyPI blog <https://blog.pypi.org>`_ explaining
+  the nature of the PEP and its behavioral implications for PyPI.
+* Updates to the `PyPI user documentation <https://docs.pypi.org/>`_ explaining
+  the difference between deletion and yanking and the limited conditions under
+  which the former can still be initiated by package owners.
```

Comment:
At least initially, it might be useful to have something "in your face" (e.g. an announcement banner?) -- most developers releasing packages won't read the PEPs, or Discourse, or the PyPI blog -- and may be surprised to learn that their ability to delete files is restricted. This is especially the case given that (I contend) uploads largely happen without using/signing in to the PyPI website -- they're via CI or the CLI.

You probably intend to do this, but it might be worth mentioning that the aforementioned blog posts will contain the motivation behind this change.